compiler/rustc_middle/src/mir/interpret/allocation/init_mask.rs

```diff
@@ -185,7 +193,7 @@ impl InitMask {
 // Note: for performance reasons when interning, some of the fields can be partially
 // hashed. (see the `Hash` impl below for more details), so the impl is not derived.
 #[derive(Clone, Debug, Eq, PartialEq, HashStable)]
-struct InitMaskMaterialized {
+pub struct InitMaskMaterialized {
```

Comment:
keep this private. Instead add an iterator over `u64` that allows getting the blocks out directly (and just produces the same value over and over in case it was lazy). If you need to box those iterators, that's fine.

Reply:
We can just store a `Vec<u64>` in SMIR, we don't need to use this optimization for now.

Reply:
> keep this private.

if I make it private I would have to remove `Stable` implementation for it.

Reply:
you can't, but you can add a function returning the iterator I described to `InitMask`

Reply:
> keep this private. Instead add an iterator over `u64` that allows getting the blocks out directly (and just produces the same value over and over in case it was lazy). If you need to box those iterators, that's fine.

but don't we lose information if it's `InitMaskBlocks::Lazy{true}` or `InitMaskBlocks::Lazy(false)`,  how would someone know if it's lazy or not, since I can't match private value. Maybe something like this ? So people who use smir could identify it it's by checking repeating 0's or 1's
```rust
vec![1;10].iter() // lazy ?
vec![0;10].iter() // not lazy ?
```

Reply:
Lazy is not information anyone cares about. It is just an internal representation detail. it's a optimization to avoid allocating a vec if all entries are true or all entries are false

Reply:
Ah I get it now, I didn't know it wasn't useful for smir